### ISSUE TITLE
PyInstaller.compat.FileNotFoundError

### ISSUE THREAD

**Issue by choidamdam** [pyinstaller/pyinstaller#1928]
`PyInstaller.compat.FileNotFoundError` occurs when a project have dependency for `pyexcelerate`. Maybe the filename `[Content_Types].xml` caused glob related error.
- Windows 10 64bit
- Python 2.7.11

Steps to reproduce:

```
$ pip install pyexcelerate
$ echo 'import pyexcelerate' > test.py
$ pyinstaller test.py
```

```
Traceback (most recent call last):
  File "C:\tools\python2\Scripts\pyinstaller-script.py", line 9, in <module>
    load_entry_point('PyInstaller==3.1.1', 'console_scripts', 'pyinstaller')()
  File "c:\tools\python2\lib\site-packages\PyInstaller\__main__.py", line 90, in run
    run_build(pyi_config, spec_file, **vars(args))
  File "c:\tools\python2\lib\site-packages\PyInstaller\__main__.py", line 46, in run_build
    PyInstaller.building.build_main.main(pyi_config, spec_file, **kwargs)
  File "c:\tools\python2\lib\site-packages\PyInstaller\building\build_main.py", line 755, in main
    build(specfile, kw.get('distpath'), kw.get('workpath'), kw.get('clean_build'))
  File "c:\tools\python2\lib\site-packages\PyInstaller\building\build_main.py", line 701, in build
    exec(text, spec_namespace)
  File "<string>", line 16, in <module>
  File "c:\tools\python2\lib\site-packages\PyInstaller\building\build_main.py", line 212, in __init__
    self.__postinit__()
  File "c:\tools\python2\lib\site-packages\PyInstaller\building\datastruct.py", line 183, in __postinit__
    self.assemble()
  File "c:\tools\python2\lib\site-packages\PyInstaller\building\build_main.py", line 434, in assemble
    imphook_object.update_dependencies(self.graph)
  File "c:\tools\python2\lib\site-packages\PyInstaller\building\imphook.py", line 309, in update_dependencies
    self._process_datas(mod_graph)
  File "c:\tools\python2\lib\site-packages\PyInstaller\building\imphook.py", line 285, in _process_datas
    self.datas.update(set(format_binaries_and_datas(self._module.datas)))
  File "c:\tools\python2\lib\site-packages\PyInstaller\building\utils.py", line 426, in format_binaries_and_datas
    src_root_path_or_glob))
PyInstaller.compat.FileNotFoundError: Path or glob "c:\tools\python2\lib\site-packages\pyexcelerate\templates\[Content_Types].xml" not found or matches no files.
```

**Comment by choidamdam** [comment]
```
PS C:\tools\python2\lib\site-packages\pyexcelerate\templates> python
Python 2.7.11 (v2.7.11:6d1b6a68f775, Dec  5 2015, 20:40:30) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import glob
>>> glob.glob('*.xml')
['[Content_Types].xml']
>>> glob.glob('[Content_Types].xml')
[]
```

**Comment by htgoebel** [comment]
Which version of PyInstaller are you using? I need to spot the exact line causing this error.

**Comment by htgoebel** [comment]
Solution could be to simply is `os.path.isfile(...)` prior to globbing it if glob returns an empty list.

**Comment by choidamdam** [comment]
@htgoebel:
The version of PyInstaller is 3.1.1.
In my local environment, I did exactly what you metioned. I've changed `lib\site-packages\PyInstaller\building\utils.py` line 421 from:
```python
        src_root_paths = glob.glob(src_root_path_or_glob)
```
to:
```python
        if os.path.exists(src_root_path_or_glob):
          src_root_paths = [src_root_path_or_glob]
        else:
          src_root_paths = glob.glob(src_root_path_or_glob)
```
then it just works fine as I expected.

**Comment by htgoebel** [comment]
Duplicate of #2314.
This was already fixed in 65927314e09e2fbadcf282e5f9881b6f231e2b0e.
